kornia/geometry/line.py

```diff
@@ -0,0 +1,173 @@
+
+        Example:
+            >>> o = torch.tensor([0.0, 0.0])
+            >>> d = torch.tensor([1.0, 1.0])
+            >>> l = ParametrizedLine(o, d)
+        """
+        super().__init__()
+        self._origin = Parameter(origin)
+        self._direction = Parameter(direction)
+
+    def __str__(self) -> str:
+        return f"Origin: {self.origin}\nDirection: {self.direction}"
+
+    def __repr__(self) -> str:
+        return str(self)
+
+    @property
+    def origin(self) -> Tensor:
+        """Return the line origin point."""
+        return self._origin
+
+    @property
+    def direction(self) -> Tensor:
+        """Return the line direction vector."""
+        return self._direction
+
+    def dim(self) -> int:
+        """Return the dimension in the line holds."""
+        return len(self.direction)
+
+    @classmethod
+    def through(cls, p0, p1) -> "ParametrizedLine":
+        """Constructs a parametrized line going from a point :math:`p0` to :math:`p1`.
+
+        Args:
+            p0: tensor with first point :math:`(B, D)`.
+            p1: tensor with second point :math:`(B, D)`.
+
+        Example:
+            >>> p0 = torch.tensor([0.0, 0.0])
+            >>> p1 = torch.tensor([1.0, 1.0])
+            >>> l = ParametrizedLine.through(p0, p1)
+        """
+        return ParametrizedLine(p0, normalize((p1 - p0), p=2, dim=-1))
+
+    @classmethod
+    def from_hyperplane(cls, plane: Hyperplane) -> "ParametrizedLine":
+        raise NotImplementedError(f"Plane not implemented yet {plane}.")
+
+    def point_at(self, t: Union[float, Tensor]) -> Tensor:
+        """The point at :math:`t` along this line.
+
+        Args:
+            t: step along the line.
+
+        Return:
+            tensor with the point.
+
+        Example:
+            >>> p0 = torch.tensor([0.0, 0.0])
+            >>> p1 = torch.tensor([1.0, 1.0])
+            >>> l = ParametrizedLine.through(p0, p1)
+            >>> p2 = l.point_at(0.1)
+        """
+        return self.origin + self.direction * t
+
+    def projection(self, point: Tensor) -> Tensor:
+        """Return the projection of a point onto the line.
+
+        Args:
+            point: the point to be projected.
+        """
+        return self.origin + (self.direction @ (point - self.origin)) * self.direction
+
+    def squared_distance(self, point: Tensor) -> Tensor:
+        """Return the squared distance of a point to its projection onte the line.
+
+        Args:
+            point: the point to calculate the distance onto the line.
+        """
+        diff: Tensor = point - self.origin
+        return squared_norm(diff - (self.direction @ diff) * self.direction)
+
+    def distance(self, point: Tensor) -> Tensor:
+        """Return the distance of a point to its projections onto the line.
+
+        Args:
+            point: the point to calculate the distance into the line.
+        """
+        return self.squared_distance(point).sqrt()
```

Comment:
squared_distance is quite innecifient used in this way. The funciton itself first compute the norm, then it is squared and now sqrt.... use directly the `norm` here.